### ISSUE TITLE
configure_optimizers documentation contradicts demo code

### ISSUE THREAD

**Issue by RylanSchaeffer** [Lightning-AI/lightning#14766]
The [documentation](https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html) for `configure_optimizers` states that two acceptable return types are:

- Dictionary, with an "optimizer" key, and (optionally) a "lr_scheduler" key whose value is a single LR scheduler or lr_scheduler_config.
- Tuple of dictionaries as described above, with an optional "frequency" key.

However, the demo code shows that the `lr_scheduler` dictionary's value needs to correspond to a dictionary, not a scheduler!

```
def configure_optimizers(self):
    optimizer1 = Adam(...)
    optimizer2 = SGD(...)
    scheduler1 = ReduceLROnPlateau(optimizer1, ...)
    scheduler2 = LambdaLR(optimizer2, ...)
    return (
        {
            "optimizer": optimizer1,
            "lr_scheduler": {
                "scheduler": scheduler1,
                "monitor": "metric_to_track",
            },
        },
        {"optimizer": optimizer2, "lr_scheduler": scheduler2},
    )

```

Can this ambiguity please be clarified?

**Comment by RylanSchaeffer** [comment]
I think this also causes a bug. If you follow the documentation and try setting the `lr_scheduler` key's value to a `scheduler` object, I don't think the scheduler is stepped.

**Comment by RylanSchaeffer** [comment]
Evidence:

![image](https://user-images.githubusercontent.com/8942987/190921992-fb085f6e-b3d1-454b-9b22-66789e715e4b.png)

**Comment by carmocca** [comment]
Hi!

I'm not seeing what's the contradiction. The statement in the docs is correct, example:

```python
import os

import torch
from torch.optim import Adam
from torch.optim.lr_scheduler import StepLR
from torch.utils.data import DataLoader

from pytorch_lightning import LightningModule, Trainer
from pytorch_lightning.demos.boring_classes import RandomDataset


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx, optimizer_idx=0):
        return self(batch).sum()

    def configure_optimizers(self):
        optimizer1 = Adam(self.layer.parameters())
        optimizer2 = Adam(self.layer.parameters())
        scheduler1 = StepLR(optimizer1, 2)
        scheduler2 = StepLR(optimizer2, 1)
        # a tuple of:
        return (
            # lr scheduler config:
            {"optimizer": optimizer1, "lr_scheduler": {"scheduler": scheduler1}},
            # lr scheduler instance:
            {"optimizer": optimizer2, "lr_scheduler": scheduler2},
        )


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=5,
        num_sanity_val_steps=0,
        limit_val_batches=0,
        max_epochs=2,
        enable_model_summary=False,
        enable_checkpointing=False,
        enable_progress_bar=False,
    )
    trainer.fit(model, train_dataloaders=train_data)
    
    print(trainer.lr_scheduler_configs)
    print(trainer.lr_scheduler_configs[0].scheduler._step_count)
    print(trainer.lr_scheduler_configs[1].scheduler._step_count)


if __name__ == "__main__":
    run()
```

And the prints confirm that they were stepped

**Comment by RylanSchaeffer** [comment]
Yeah I agree. I misread the documentation and your example shows that. I'll investigate.
